packages/@aws-cdk-testing/cli-integ/lib/shell.ts

```diff
@@ -27,37 +29,54 @@ export async function shell(command: string[], options: ShellOptions = {}): Prom
 
   const env = options.env ?? (options.modEnv ? { ...process.env, ...options.modEnv } : process.env);
 
-  const child = child_process.spawn(command[0], command.slice(1), {
+  // copy because we will be shifting it
+  const interact = [...(options.interact ?? [])];
+
+  const child = Process.spawn(command[0], command.slice(1), {
     ...options,
     env,
-    // Need this for Windows where we want .cmd and .bat to be found as well.
-    shell: true,
-    stdio: ['ignore', 'pipe', 'pipe'],
-  });
+    tty: interact.length > 0,
+  })
 
   return new Promise<string>((resolve, reject) => {
     const stdout = new Array<Buffer>();
     const stderr = new Array<Buffer>();
 
-    child.stdout!.on('data', chunk => {
+    const stdoutSinceLastPrompt = new Array<Buffer>();
+
+    child.onStdout(chunk => {
       if (show === 'always') {
-        writeToOutputs(chunk);
+        writeToOutputs(chunk.toString('utf-8'));
       }
       stdout.push(chunk);
+      stdoutSinceLastPrompt.push(chunk);
+
+      const interaction = interact[0];
+      if (interaction) {
+
+        if (interaction.prompt.test(Buffer.concat(stdoutSinceLastPrompt).toString('utf-8'))) {
+          // subprocess expects a user input now
+          child.writeStdin(interaction.input + os.EOL);
```

Comment:
Do we definitely know the input must always be terminated by a newline? Nothing will ever just respond to single keypresses?

Reply:
I see that interactions must happen in order. 

Should it be an error if we didn't consume all interactions when the process exits?

Reply:
Yeah makes sense, i'll add it.

Reply:
> Do we definitely know the input must always be terminated by a newline? Nothing will ever just respond to single keypresses?

I think `promptly` (which is what we use in the CLI) responds to EOL. But that's also probably a configuration option...So no, not definite...I guess we'll find out when we add more tests.

Do you prefer to just move the `EOL` to the interaction input, so its controlled by the test itself?

Reply:
I'll just give it as an option of `UserInteraction`

Reply:
Addressed